cmd/erasure-zones.go

```diff
@@ -542,14 +543,14 @@ func (z *erasureZones) DeleteObject(ctx context.Context, bucket string, object s
 func (z *erasureZones) DeleteObjects(ctx context.Context, bucket string, objects []ObjectToDelete, opts ObjectOptions) ([]DeletedObject, []error) {
 	derrs := make([]error, len(objects))
 	dobjects := make([]DeletedObject, len(objects))
-	objNames := make([]string, len(objects))
+	objSets := set.NewStringSet()
 	for i := range derrs {
 		derrs[i] = checkDelObjArgs(ctx, bucket, objects[i].ObjectName)
-		objNames[i] = objects[i].ObjectName
+		objSets.Add(objects[i].ObjectName)
```

Comment:
So both requests will go through?

With versioning where things are not actually deleted that seems problematic.

Shouldn't the second one be rejected?

Reply:
what both requests? we only hold locks for object name @klauspost if things can deleted or not is already filtered in the top layer